docs/04.guides/04.cookbooks/24.Query-Handling/page.md

```diff
@@ -0,0 +1,250 @@
+---
+title: Query Handling In Lucee
+id: query-handling-lucee
+---
+
+Here we can find how the lucee supports for SQL queries. 
+
+## Query tags ##
+
+```<cfquery>``` different ways to use the tags in lucee & how we can the passed the value into the query
+
+```lucee
+<cfquery name="qry" datasource="test">
+	select * from Foo1890
+</cfquery>
+<cfdump var="#qry#" expand="false">
+```
+
+Above example just retrieve data from database.
+
+
+### Using QueryParam ###
+
+Queryparam used inside the query tag, It used to bind the value with sql statement
+
+```lucee
+<cfquery name="qry" datasource="test">
+	select * from Foo1890
+	where title=<cfqueryparam sqltype="varchar" value="Susi">
+</cfquery>
+<cfdump var="#qry#" expand="false">
+```
+
+Passing value with QueryParam, has two advantage,
+
+* Value you passed in queryparam is very secured,
+* Lucee able to cache the statement query statement & reused it as long the value changed.
+
+### params ###
+Here we use param as part of cfquery tag, used to pass the value with SQL statement.
+
+Pass the params value with struct
+
+```lucee
+<cfquery name="qry" datasource="test" params="#{title:'Susi'}#">
+	select * from Foo1890
+	where title=:title
+</cfquery>
+<cfdump var="#qry#" expand="false">
+```
+
+referenced as ```:key``` in sql.
+
+Below example to pass the more information with struct
+
+```lucee
+<cfquery name="qry" datasource="test" params="#{title:{type:'varchar',value:'Susi'}}#">
+	select * from Foo1890
+	where title=:title
+</cfquery>
+<cfdump var="#qry#" expand="false">
+```
+
+You can passed the params value by array, It referenced as `?` in SQL
+
+```lucee
+<cfquery name="qry" datasource="test" params="#['Susi']#">
+	select * from Foo1890
+	where title=?
+</cfquery>
+<cfdump var="#qry#" expand="false">
+```
+
+Below example to pass the more information with array
+
+```lucee
+<cfquery name="qry" datasource="test" params="#[{type:'varchar',value:'Susi'}]#">
+	select * from Foo1890
+	where title=?
+</cfquery>
+```
+
+## Query with script tag ##
+
+Lucee supports for two types script statements like below,
+
+```lucee
+query name="qry" datasource="test" params={title:{type:'varchar',value:'Susi'}} {
+	echo("
+		select * from Foo1890
+		where title=:title
+	");
+}
+dump(qry);
+```
+
+```lucee
+cfquery(name="qry",datasource="test",params={title:{type:'varchar',value:'Susi'}}) {
+	echo("
+		select * from Foo1890
+		where title=:title
+	");
+}
+dump(qry);
+```
+
+## QueryExecute ##
+
+It supports for both script tag & regular tag, you can pass all the arguments to the function.
+
+```lucee
+qry=queryExecute(
+	sql:"select * from Foo1890 where title=:title"
+	,options:{datasource="test"}
+	,params={title:{type:'varchar',value:'Susi'}}
+);
+dump(qry);
+```
+
+Pass the values in params same as we saw in cfquerytag, In options we can pass other arguments like datasource,cachename,dbtype
```

Comment:
***toChange***
***I think it would be helpful to tell developers where they can find the full list of options - please replace my ??? with this informaiton.***
The params are passed in the same way we saw in cfquerytag.
In the options section, we can pass in other arguments like datasource, cachename, dbtype, etc. For a full list of the options that can be passed in see ???